ocrd/ocrd/mets_server.py

```diff
@@ -0,0 +1,253 @@
+"""
+# METS server functionality
+"""
+import re
+from os import environ, _exit
+from io import BytesIO
+from typing import Any, Dict, Optional, Union, List, Tuple
+
+from fastapi import FastAPI, Request, File, Form, UploadFile
+from fastapi.responses import JSONResponse
+from requests import request, Session as requests_session
+from requests_unixsocket import Session as requests_unixsocket_session
+from pydantic import BaseModel, Field, constr, ValidationError
+
+import uvicorn
+
+from ocrd_models import OcrdMets, OcrdFile
+from ocrd_utils import initLogging, getLogger, deprecated_alias
+
+#
+# XXX HACKS TODO
+#
+initLogging()
+
+#
+# Models
+#
+
+class OcrdFileModel(BaseModel):
+    file_grp : str = Field()
+    file_id : str = Field()
+    mimetype : str = Field()
+    page_id : Union[str, None] = Field()
+    url : str = Field()
+
+    @staticmethod
+    def create(file_grp : str, file_id : str, page_id : Union[str, None], url : str):
+        return OcrdFileModel(file_grp=file_grp, file_id=file_id, page_id=page_id, mimetype=mimetype, url=url)
+
+class OcrdAgentModel(BaseModel):
+    name : str = Field()
+    _type : str = Field()
+    role : str = Field()
+    otherrole : str = Field()
+    othertype : str = Field()
+    notes : List[Tuple[Dict[str, str], Optional[str]]] = Field()
+
+    @staticmethod
+    def create(name : str, _type : str, role : str, otherrole : str, othertype : str, notes : List[Tuple[Dict[str, str], Optional[str]]]):
+        return OcrdAgentModel(name=name, _type=_type, role=role, otherrole=otherrole, othertype=othertype, notes=notes)
+
+
+class OcrdFileListModel(BaseModel):
+    files : List[OcrdFileModel] = Field()
+
+    @staticmethod
+    def create(files : List[OcrdFile]):
+        return OcrdFileListModel(
+            files=[OcrdFileModel(file_grp=f.fileGrp, file_id=f.ID, mimetype=f.mimetype, page_id=f.pageId, url=f.url) for f in files]
```

Comment:
```suggestion
            files=[OcrdFileModel.create(file_grp=f.fileGrp, file_id=f.ID, mimetype=f.mimetype, page_id=f.pageId, url=f.url) for f in files]
```

The rest of the models seem good.